src/components/table/table-header/table-header.js

```diff
@@ -96,31 +104,73 @@ class TableHeader extends React.Component {
   }
 
   /**
-   * Emits sort event to parent context - table.
+   * Event handler for clicks on the <a> tag used for activating
+   * column sorting.
    *
-   * @method emitSortEvent
+   * Used to prevent the default action of the <a> tag.
+   *
+   * @method onSortableColumnClick
+   * @return {Void}
    */
-  emitSortEvent = () => {
-    let sortOrder = this.context.sortOrder || 'desc';
+  onSortableColumnClick(event) {
+    event.preventDefault();
+  }
 
-    // If this is the current sorted column. flip order
-    if (this.sorted) {
-      sortOrder = this.context.sortOrder === 'asc' ? 'desc' : 'asc';
-    }
+  /**
+   * Returns props to be used on the TH element.
+   *
+   * @method tableHeaderProps
+   * @return {Object}
+   */
+  get tableHeaderProps() {
+    const { ...props } = validProps(this);
 
-    this.context.onSort(this.props.name, sortOrder);
+    delete props.children;
+
+    props.className = this.tableHeaderClasses();
+    props.onClick = this.props.sortable ? this.emitSortEvent.bind(this) : '';
```

Comment:
I realise this is not your code, but is the `.bind` actually necessary here? `emitSortEvent` is defined using the fat arrow syntax.